### PR TITLE
Init project and CI

### PR DESCRIPTION
**Pull request by MarekSuchanek**
- Setup CI (including code style and security checks)
- Basic CRUD REST API for Train Garages and Station Directories (traditional Spring Boot stack with PostgreSQL DB)
- Auto-generated Swagger UI for OpenAPI
- Basic tests for CRUD REST API, Swagger UI, and Actuator Info endpoint

To be **squash-merged**.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  pull_request:
+
+jobs:
+
+  test:
+    name: Maven Test & Package
+    runs-on: ubuntu-latest
+
+    env:
+      JAVA_DISTRIBUTION: microsoft
```
Comment (kburger) [inline .github/workflows/build.yml]: Any reason for the micosoft distribution vs [temurin](https://github.com/actions/setup-java/blob/main/README.md#Supported-distributions)?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-${{ env.JAVA_VERSION }}-maven-${{ hashFiles('**/pom.xml') }}
+
+      # (2) -> Prepare Java
+      - name: Setup Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: ${{ env.JAVA_DISTRIBUTION }}
+          java-version: ${{ env.JAVA_VERSION }}
```
Comment (kburger) [inline .github/workflows/build.yml]: Cache can be simplified with newer versions of setup-java I believe. See https://github.com/actions/setup-java/blob/main/README.md#caching-maven-dependencies for more info.

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+		<finalName>app</finalName>
+		<plugins>
+			<plugin>
+				<groupId>org.springframework.boot</groupId>
+				<artifactId>spring-boot-maven-plugin</artifactId>
+			</plugin>
+			<plugin>
+				<groupId>com.mycila</groupId>
+				<artifactId>license-maven-plugin</artifactId>
+				<version>${plugin.license.version}</version>
```
Comment (kburger) [inline pom.xml]: license plugin has a newer version (and with that a breaking change in the configuration) available IIRC.

File: src/main/java/org/fairdatatrain/trainhandler/service/actuator/AppInfoContributor.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    private String buildTime;
+
+    @Override
+    public void contribute(Info.Builder builder) {
+        builder.withDetail("name", "FAIR Data Point");
+        builder.withDetail("version", getFdpVersion());
+        builder.withDetail("builtAt", buildTime);
+    }
+
+    public String getFdpVersion() {
```
Comment (kburger) [inline src/main/java/org/fairdatatrain/trainhandler/service/actuator/AppInfoContributor.java]: `Fdp`?

**Review by kburger — APPROVED** [review]

File: pom.xml (hunk 3 above)
Comment (MarekSuchanek) [reply pom.xml]: Seems like there is just RC for 4.2: https://mvnrepository.com/artifact/com.mycila/license-maven-plugin

File: .github/workflows/build.yml (hunk 1 above)
Comment (MarekSuchanek) [reply .github/workflows/build.yml]: Not really, GitHub is Microsoft, Actions run on Azure, etc. 😄 But we can decide to use Temurin... Then, I would also use `maven:3.8.4-eclipse-temurin-17` and `eclipse-temurin:17` as base in Dockerfiles.

File: src/main/java/org/fairdatatrain/trainhandler/service/actuator/AppInfoContributor.java (hunk 4 above)
Comment (MarekSuchanek) [reply src/main/java/org/fairdatatrain/trainhandler/service/actuator/AppInfoContributor.java]: Thanks!

**Pull request merged**